src/posix/platform/rcp_caps_diag.cpp

```diff
@@ -498,22 +502,61 @@ void RcpCapsDiag::SetDiagOutputCallback(otPlatDiagOutputCallback aCallback, void
     mOutputContext  = aContext;
 }
 
+void RcpCapsDiag::ProcessPerf(void) { TestNumSupportedSleepyChildren(); }
+
+void RcpCapsDiag::TestNumSupportedSleepyChildren(void)
+{
+    static constexpr uint16_t kMaxNumChildren = 512;
+    static constexpr uint16_t kValueLength    = 6;
+    otExtAddress              extAddress      = {0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, 0x88};
+    char                      value[kValueLength];
+    uint16_t                  num = 0;
+
+    SuccessOrExit(mRadioSpinel.Set(SPINEL_PROP_MAC_SRC_MATCH_EXTENDED_ADDRESSES, nullptr));
+
+    for (num = 0; num < kMaxNumChildren; num++)
+    {
+        *reinterpret_cast<uint16_t *>(extAddress.m8) = num;
+        SuccessOrExit(
+            mRadioSpinel.Insert(SPINEL_PROP_MAC_SRC_MATCH_EXTENDED_ADDRESSES, SPINEL_DATATYPE_EUI64_S, extAddress.m8));
```

Comment:
This indirectly assumes that the size of the `SPINEL_PROP_MAC_SRC_MATCH_EXTENDED_ADDRESSES` array in the radio indicates the maximum number of sleepy children that can be supported.

However, this assumption may not always hold true. A radio platform could opt to use a smaller source-match table while the OpenThread stack could still function and support a larger number of sleepy children than the maximum size of the source-match table.

Reply:
Agree with you. I have changed the code to check the source match table size. So there won't be the assumption now.